extension/ts/copy.ts

```diff
@@ -0,0 +1,90 @@
+import { RELEVANT_TAGS } from "./constants";
+import { SaveData } from "./types";
+import { getElementsByTags } from "./util";
+
+const COLLECTIONS_ID_PREFIX = "collection_u_";
+const SAVE_DATA_KEY = "_save-data";
+const COLLECTIONS_KEY = "___collections_";
+
+const saveData = (parent: HTMLElement) => (event: Event) => {
+	event.preventDefault();
+	localStorage.setItem(SAVE_DATA_KEY, JSON.stringify(getSaveData(parent)));
+};
+
+const loadData = (parent: HTMLElement) => (event: Event) => {
+	event.preventDefault();
+	const saveData = JSON.parse(localStorage.getItem(SAVE_DATA_KEY) ?? "{}");
+	insertSaveData(parent, saveData);
+};
+
+const getSaveData = (parent: HTMLElement) => {
+	const elements = getElementsByTags(parent, RELEVANT_TAGS);
+	return elements.reduce((saveData: SaveData, element: any) => {
+		// We can't change hidden elements because LibraryThing relies
+		// on hidden form inputs to send additional, form-specific metadata
+		// on save
+		if (element && element.id && element.type !== "hidden") {
```

Comment:
super nit: i think `element && element.id` can be made shorter by doing `!!element?.id`

Reply:
`!!` not needed.
```typescript
if (element?.id && element.type !== "hidden") {
```

Reply:
I've been thinking about this, and I think I prefer it with the explicit `element`.

If it were
```typescript
if (element?.id) {
```
I'd say that that's _perfect_. 

Reasoning:

`element?.id` accomplishes two checks:
1. `element` is truthy
1. `id` is truthy

However, `element.type` depends on the check "`element` is truthy", which is hidden inside the `element?.id`, and contains a check that is unrelated to `element.type`: "`id` is truthy".

You could put the "`element` is truthy" inside both expressions, however `element?.type !== "hidden"` on its own would be a bug, so I don't think that's the right way.

Reply:
(I often think shorter is bad for readability and maintainability)